corehq/apps/hqwebapp/models.py

```diff
@@ -369,8 +369,11 @@ class DashboardTab(UITab):
 
     @property
     def is_viewable(self):
-        return (self.domain and self.project and not self.project.is_snapshot
-                and self.couch_user)
+        return (self.domain and self.project and
+                not self.project.is_snapshot and
+                self.couch_user and
+                # domain hides Dashboard tab if user is non-admin
+                not (toggles.CUSTOM_MENU_BAR.enabled(self.domain) and not self.couch_user.is_superuser))
```

Comment:
Would have been neater if this were to go into custom domain specific modules. but they prefer a one-off feature toggle instead. (see the FB ticket for more)